roles/ec2_instance_create_delete/README.md

```diff
@@ -0,0 +1,171 @@
+  The instance type for the EC2 instance (e.g., `t2.micro`, `m5.large`).
+
+* **ec2_instance_create_delete_ami_id**: (Required)
+  The AMI ID for the EC2 instance.
+
+* **ec2_instance_create_delete_key_name**: (Optional)
+  The name of the key pair to use for SSH access to the EC2 instance.
+  If the key does not exist, a key pair will be created with the name.
+
+* **ec2_instance_create_delete_vpc_subnet_id**: (Optional)
+  The ID of the VPC subnet in which the instance will be launched.
+  If not provided, instance might get created with `default` subnet in the AWS region if present.
+
+* **ec2_instance_create_delete_tags**: (Optional)
+  A dictionary of tags to assign to the EC2 instance.
+
+* **ec2_instance_create_delete_wait_for_boot**: (Optional)
+  Whether to wait for the EC2 instance to be in the "running" or "terminated" state before continuing. Default is `true`.
+
+### Optional Networking Resources
+
+#### Elastic IP
+
+* **ec2_instance_create_delete_vpc_id**: (Optional)
+  The ID of the VPC used for security group and internet gateway.
+  Required if `ec2_instance_create_delete_associate_igw` or `ec2_instance_create_delete_associate_eip` is `true`.
+
+* **ec2_instance_create_delete_associate_eip**: (Optional)
+  Whether to create an Elastic IP (EIP) and associate it with the EC2 instance. Default is `false`.
+  If set to `true` and the provided VPC doesn't have an Internet Gateway (IGW) attached, set `ec2_instance_create_delete_associate_igw` to `true` to avoid failure.
+
+* **ec2_instance_create_delete_eip_tags**: (Optional)
+  Tags to assign to the elastic IP.
+
+#### Internet Gateway
+
+* **ec2_instance_create_delete_associate_igw**: (Optional)
+  Whether to create and associate an internet gateway with the EC2 instance. Default is `false`.
+  If set to `true`, an internet gateway will be created or associated with the instance.
+
+* **ec2_instance_create_delete_igw_tags**: (Optional)
+  Tags to assign to the internet gateway.
+
+#### External Security Group
+
+* **ec2_instance_create_delete_associate_external_sg**: (Optional)
+  Whether to create and associate an security group with the EC2 instance. Default is `false`.
+  If set to `true`, an security group will be created or associated with the instance.
+
+* **ec2_instance_create_delete_external_sg_name**: (Required)
+  The name of the security group to use for the EC2 instance.
+  The role will check if an SG with this name exists. If not, it will create a new one.
+  Default is `ec2_instance_create-default-external-sg`.
+
+* **ec2_instance_create_delete_external_sg_description**: (Optional)
+  A description for the security group. Default is `Security group for external access`.
+
+* **ec2_instance_create_delete_external_sg_rules**: (Optional)
+  A list of custom rules to add to the security group. Each rule is a dictionary with `proto`, `ports`, and `cidr_ip` keys. Default is to allow SSH (port 22) from `0.0.0.0/0`.
```

Comment:
Let's not default to this. Probably best to just not include any inbound rules by default?